assets/js/modules/tagmanager/setup.js

```diff
@@ -441,36 +434,23 @@ class TagmanagerSetup extends Component {
 			onSettingsPage,
 		} = this.props;
 
-		if ( ! errorCode ) {
+		if ( 0 === errorMsg.length ) {
 			return null;
 		}
 
-		let showError = true; // default error message.
-		let showNotice = false;
-		let message = errorMsg;
-
-		switch ( true ) {
-				case onSettingsPage && 'insufficientPermissions' === errorCode:
-					showError = false;
-					showNotice = true;
-					break;
-		}
-
-		if ( showError && 0 < message.length ) {
-			return (
-				<div className="googlesitekit-error-text">
-					<p>{ __( 'Error:', 'google-site-kit' ) } { message }</p>
-				</div>
-			);
-		}
+		const showErrorFormat = onSettingsPage && 'insufficientPermissions' === errorCode ? false : true; // default error format.
 
-		if ( showNotice && 0 < message.length ) {
-			return (
-				<div>
-					<p>{ message }</p>
-				</div>
-			);
-		}
+		return (
+			<div className={ showErrorFormat ? 'googlesitekit-error-text' : '' }>
+				<p>{
+					sprintf(
+						'%s %s',
+						showErrorFormat ? __( 'Error:', 'google-site-kit' ) : '',
+						errorMsg
+					)
+				}</p>
+			</div>
+		);
```

Comment:
What I meant here was more like this:

```jsx
<p>
    {
        showErrorFormat
            /* translators: ... */
            ? sprintf( __( 'Error: %s', 'google-site-kit' ), errorMsg )
            : errorMsg
    }
</p>
```

or the message logic could be handled above the JSX and this part could be simplified to `<p>{message}</p>` - whichever you feel is more readable.

The important change here though is un-concatenating the translated string.